integration/src/test/resources/non-blocking-test-config.xml

```diff
@@ -251,6 +253,52 @@
         </foreach>
     </flow>
 
+    <flow name="untilSuccessful" initialState="stopped">
+        <test:assert-intercepting responseSameTask="false"/>
+        <until-successful>
+            <test:assert-intercepting responseSameTask="false" count="2"/>
+            <test:non-blocking-processor/>
+            <until-successful>
+                <test:assert-intercepting responseSameTask="false" count="6"/>
+                <test:non-blocking-processor/>
+            </until-successful>
+        </until-successful>
+        <test:assert expression="#[payload == 'Test Message']"/>
+    </flow>
+
+    <flow name="foreach_and_untilSuccessful" initialState="stopped">
+        <test:assert-intercepting responseSameTask="false"/>
+        <foreach>
+            <test:assert-intercepting responseSameTask="false" count="2"/>
+            <test:non-blocking-processor/>
+            <until-successful>
+                <test:assert-intercepting responseSameTask="false" count="6"/>
+                <test:non-blocking-processor/>
+            </until-successful>
+        </foreach>
+    </flow>
+
+    <flow name="untilSuccessful_and_foreach" initialState="stopped">
+        <test:assert-intercepting responseSameTask="false"/>
+        <set-payload value="#[mel: [ 'A', 'B' ]  ]"/>
```

Comment:
use DW instead of MEL. Now that you're at it, migrate all other MEL usages in this file to mel